lib/imagery/testsuite/test_imagery_signature_management.py

```diff
@@ -0,0 +1,702 @@
+        # As signatures are created directly not via signature creation
+        # tools, we must ensure signature directories exist
+        os.makedirs(cls.mpath + "/signatures/sig/", exist_ok=True)
+        os.makedirs(cls.mpath + "/signatures/sigset/", exist_ok=True)
+        # A mapset with a random name
+        cls.src_mapset_name = tempname(10)
+        G_make_mapset(None, None, cls.src_mapset_name)
+        cls.src_mapset_path = (
+            cls.mpath.rsplit("/", maxsplit=1)[0] + "/" + cls.src_mapset_name
+        )
+        os.makedirs(cls.src_mapset_path + "/signatures/sig/")
+        cls.src_sig = tempname(10)
+        cls.sigfiles.append(
+            "{}/signatures/sig/{}".format(cls.src_mapset_path, cls.src_sig)
+        )
+        f = open(cls.sigfiles[0], "w")
+        f.write("A sig file")
+        f.close()
+        os.makedirs(cls.src_mapset_path + "/signatures/sigset/")
+        cls.src_sigset = tempname(10)
+        cls.sigfiles.append(
+            "{}/signatures/sigset/{}".format(cls.src_mapset_path, cls.src_sigset)
+        )
+        f = open(cls.sigfiles[1], "w")
+        f.write("A sigset file")
+        f.close()
+
+    @classmethod
+    def tearDownClass(cls):
+        # Remove random mapset created during setup
+        shutil.rmtree(cls.src_mapset_path, ignore_errors=True)
+        for f in cls.sigfiles:
+            try:
+                os.remove(f)
+            except OSError:
+                pass
+
+    def test_copy_to_wrong_mapset(self):
+        rnd_name = "{0}@{0}".format(tempname(10))
+        ret = I_signatures_copy(
+            SIGFILE_TYPE_SIG, tempname(10), self.mapset_name, rnd_name
+        )
+        self.assertEqual(ret, 1)
+
+    def test_sig_does_not_exist(self):
+        ret = I_signatures_copy(
+            SIGFILE_TYPE_SIG, tempname(10), self.mapset_name, tempname(10)
+        )
+        self.assertEqual(ret, 1)
+
+    def test_sigset_does_not_exist(self):
+        ret = I_signatures_copy(
+            SIGFILE_TYPE_SIGSET, tempname(10), self.mapset_name, tempname(10)
+        )
+        self.assertEqual(ret, 1)
+
+    def test_success_unqualified_sig(self):
+        dst = tempname(10)
+        ret = I_find_signature(SIGFILE_TYPE_SIG, dst, self.mapset_name)
+        self.assertFalse(ret)
+        ret = I_find_signature(SIGFILE_TYPE_SIG, self.src_sig, self.src_mapset_name)
+        self.assertTrue(ret)
+        ret = I_signatures_copy(
+            SIGFILE_TYPE_SIG, self.src_sig, self.src_mapset_name, dst
+        )
+        self.sigfiles.append("{}/signatures/sig/{}".format(self.mpath, dst))
+        self.assertEqual(ret, 0)
+        ret = I_find_signature(SIGFILE_TYPE_SIG, dst, self.mapset_name)
+        self.assertTrue(ret)
+        ms = utils.decode(ret)
+        self.assertEqual(ms, self.mapset_name)
+
+    def test_success_fq_sig(self):
+        dst = tempname(10)
+        self.sigfiles.append("{}/signatures/sig/{}".format(self.mpath, dst))
+        dst = dst + "@" + self.mapset_name
```

Comment:
Use f-strings for these cases too.

Reply:
Done. I f'ed up all strings in imagery library test suite.